scripts/ci-entrypoint.sh

```diff
@@ -213,33 +210,32 @@ install_addons() {
     timeout --foreground 1800 bash -c wait_for_nodes
 
     echo "Waiting for all calico-system pods to be ready"
-    "${KUBECTL}" wait --for=condition=Ready pod -n calico-system --all --timeout=10m
+    "${KUBECTL}" --kubeconfig "${PWD}/${CLUSTER_NAME}.kubeconfig" wait --for=condition=Ready pod -n calico-system --all --timeout=10m
 
     echo "Waiting for all kube-system pods to be ready"
-    "${KUBECTL}" wait --for=condition=Ready pod -n kube-system --all --timeout=10m
+    "${KUBECTL}" --kubeconfig "${PWD}/${CLUSTER_NAME}.kubeconfig" wait --for=condition=Ready pod -n kube-system --all --timeout=10m
+
+    # For convenience we export the KUBECONFIG environment variable
+    # so that future steps in a serialized workflow can target the workload cluster automatically
+    export KUBECONFIG="${PWD}/${CLUSTER_NAME}.kubeconfig"
```

Comment:
@CecileRobertMichon as I audited that many variations of k8s + Azure tests in the ecosystem that depend on this script I observed that we have essentially inherited a contract whereby we must export `KUBECONFIG` in this way to allow serialized CI jobs to target their work against the CAPZ-built Azure cluster.

Thanks @marosset for the extra eyes :)

Reply:
I wonder if we should do this outside the install_addons func then, as a final step before returning control to the external test suite (i.e. on line 280 https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3126/files#diff-2009f7be5363812e22862771f557363795759e67b3ca2e32c50d9fd5f8c5669bR274)